nethttp/client_test.go

```diff
@@ -43,14 +44,16 @@ func TestClientTrace(t *testing.T) {
 	defer srv.Close()
 
 	tests := []struct {
-		url    string
-		num    int
-		opts   []ClientOption
-		opName string
+		url            string
+		num            int
+		opts           []ClientOption
+		opName         string
+		additionalTags map[string]interface{}
 	}{
 		{url: "/ok", num: 3, opts: nil, opName: "HTTP Client"},
 		{url: "/redirect", num: 4, opts: []ClientOption{OperationName("client-span")}, opName: "client-span"},
-		{url: "/fail", num: 3, opts: nil, opName: "HTTP Client"},
+		{url: "/fail", num: 3, opts: nil, opName: "HTTP Client", additionalTags: makeTags(string(ext.Error), true)},
+		{url: "/ok", num: 3, opts: []ClientOption{ClientSpanObserver(func(s opentracing.Span, r *http.Request) { s.SetTag("hello", "world") })}, opName: "HTTP Client", additionalTags: makeTags("hello", "world")},
```

Comment:
I would suggest defining the function above the tests array, for better  readability
```
helloWorld := func(s opentracing.Span, r *http.Request)  { ... }
```